### PR TITLE
MINOR: Close create topics policy during shutdown and more tests

### PR DESCRIPTION
**Pull request by ijuma**

### CODE REVIEW

**Comment by ijuma** [comment]
Review by @hachikuji.

**Comment by ijuma** [comment]
retest this please

File: core/src/test/scala/unit/kafka/server/CreateTopicsRequestWithPolicyTest.scala [hunk 1]
```diff
@@ -65,14 +70,23 @@ class CreateTopicsRequestWithPolicyTest extends AbstractCreateTopicsRequestTest
...
+      Map("policy-topic4" -> error(Errors.POLICY_VIOLATION, Some("RetentionMs should be less than 5000ms if replicationFactor > 5"))))
+
     val assignments = replicaAssignmentToJava(Map(0 -> List(1), 1 -> List(0)))
     validateErrorCreateTopicsRequests(new CreateTopicsRequest.Builder(
-      Map("policy-topic3" -> new CreateTopicsRequest.TopicDetails(assignments)).asJava, timeout).build(),
-      Map("policy-topic3" -> error(Errors.POLICY_VIOLATION,
-        Some("""Topic partitions should have at least 2 partitions, received 1 for partition 0"""))), checkErrorMessage = true)
+      Map("policy-topic5" -> new CreateTopicsRequest.TopicDetails(assignments)).asJava, timeout).build(),
+      Map("policy-topic5" -> error(Errors.POLICY_VIOLATION,
+        Some("Topic partitions should have at least 2 partitions, received 1 for partition 0"))), checkErrorMessage = true)
```
Comment (hachikuji) [inline core/src/test/scala/unit/kafka/server/CreateTopicsRequestWithPolicyTest.scala]: Any reason to explicitly enable `checkErrorMessage` here and not the others? The default is `true`.

**Review by hachikuji — APPROVED** [review]
Trivial comment. LGTM.

File: core/src/test/scala/unit/kafka/server/CreateTopicsRequestWithPolicyTest.scala (hunk 1 above)
Comment (ijuma) [reply core/src/test/scala/unit/kafka/server/CreateTopicsRequestWithPolicyTest.scala]: Just an oversight, removed it.

**Pull request closed without merging**